### PR TITLE
Cross build with Scala 2.11.8 and 2.12.0

### PR DESCRIPTION
**Pull request by retronym**
- Workaround a Scala 2.12.0 compiler bug in the `fields` phase
    by adding a type annotation.
  - Update to recent versions of dependencies
  - Remove a rule for the experimental Akka streams API. This rule and
    test need to be reworked for the stable release of Akka streams as
    the API has changed.
  - Enable cross building using sbt-doge. This is needed because we have
    a SBT plugin subproject, which must be build once only with Scala
    2.10 even when cross building the rest with 2.11/2.12.
    The magic incantation is `sbt -J-Xmx2G ';+test;+publishLocal'`.
  - Enable cross building in the Travis CI config, using a suitable
    JDK in each case.

### CODE REVIEW

**Comment by retronym** [comment]
Review by @SethTisue

**Comment by SethTisue** [comment]
outstanding!

**Comment by retronym** [comment]
Fixes #74 #79

**Comment by retronym** [comment]
Compiler regression lodged as: https://github.com/scala/scala-dev/issues/268

File: .travis.yml [hunk 1]
```diff
@@ -2,8 +2,14 @@ sudo: false
 language: scala
 script:
   - sbt clean update compile test
```
Comment (xuwei-k) [inline .travis.yml]:
```diff
-  - sbt clean update compile test
+  - sbt sbt-abide/test ++${TRAVIS_SCALA_VERSION} abide/test abide-core/test abide-extra/test abide-macros/test abide-akka/test
```

or

```diff
-  - sbt clean update compile test
-matrix:
-  include:
-  - scala: 2.11.8
-    jdk: openjdk6
-    env: IS_PUBLISH_JDK=true
-  - scala: 2.12.0
-    jdk: oraclejdk8
-    env: IS_PUBLISH_JDK=true
+  - sbt "so test"
```

Comment (SethTisue) [inline .travis.yml]: thanks! Dale picked this up at https://github.com/scala/scala-abide/pull/80

**Pull request merged**